lib/common.ml

```diff
@@ -65,7 +65,13 @@ let longest_common_prefix xs =
   match xs with
   | [] -> ""
   | [ x ] -> x
-  | x :: _ -> List.sort (Fun.flip String.compare) xs |> List.hd |> Stre.common_prefix x |> String.sub x 0
+  | x :: xs -> List.fold_left (fun a s -> Stre.common_prefix a s |> String.sub a 0) x xs
+
+(* let longest_common_prefix xs =
```

Comment:
can remove this?
also maybe the function name is too general for its intended usecase/expected behavior

Reply:
maybe we should remove this from `lib/common.ml` entirely and just write the logic here in `lib/slack_message.ml` where it's called?

Reply:
let's do that. This whole part can be replaced I think:

```
let prefix_path =
      List.map (fun f -> f.filename) files
      |> Common.longest_common_prefix
      |> String.split_on_char '/'
      |> drop_last
      |> String.concat "/"
```

so as not to rely on assumptions about what happens to the function output like:

> we always drop the last path element